app/plugins/rwahsNavigation/rwahsNavigationPlugin.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+require_once(__CA_MODELS_DIR__ . '/ca_search_forms.php');
+require_once(__CA_MODELS_DIR__ . '/ca_bundle_displays.php');
+
+class rwahsNavigationPlugin extends BaseApplicationPlugin {
+
+    /** @var Configuration */
+    private $opo_config;
+
+    public function __construct($ps_plugin_path) {
+        parent::__construct();
+        $this->description = _t('Customises the navigation menu for Royal WA Historical Society requirements');
+        $this->opo_config = Configuration::load($ps_plugin_path . '/conf/rwahsNavigation.conf');
+    }
+
+    public function checkStatus() {
+        return array(
+            'description' => $this->getDescription(),
+            'errors' => $this->_checkNewMenuShortcuts() + $this->_checkSearchMenuShortcuts(),
+            'warnings' => array(),
+            'available' => ((bool)$this->opo_config->get('enabled'))
+        );
+    }
+
+    private function _checkNewMenuShortcuts() {
+        $va_errors = array();
+        $va_shortcuts = $this->opo_config->get('new_menu_shortcuts');
+        if (is_array($va_shortcuts)) {
+            foreach ($va_shortcuts as $vs_key => $vs_type_code) {
+                if (!is_string($vs_type_code)) {
+                    array_push($va_errors, _t('Custom new menu shortcut with key "%1" is not a type code (string value)', $vs_key));
+                }
+            }
+        }
+        return $va_errors;
+    }
+
+    private function _checkSearchMenuShortcuts() {
+        $va_errors = array();
+        $va_shortcuts = $this->opo_config->get('search_menu_shortcuts');
+        if (is_array($va_shortcuts)) {
+            foreach ($va_shortcuts as $vs_key => $va_shortcut) {
+                if (!is_array($va_shortcut)) {
+                    array_push($va_errors, _t('Custom search shortcut with key "%1" is not an array', $vs_key));
```

Comment:
No nothing wrong, I just figured `array_push` fit better with `is_array` and `isset`.